physics/GFS_rrtmg_pre.F90

```diff
@@ -871,6 +985,40 @@ subroutine GFS_rrtmg_pre_run (Model, Grid, Sfcprop, Statein,   & ! input
                          Tbd%phy_f3d(:,:,2), Tbd%phy_f3d(:,:,3),        &
                          clouds,cldsa,mtopa,mbota, de_lgth)            !  --- outputs
 
+
+        elseif(Model%imp_physics == Model%imp_physics_thompson) then                              ! Thompson MP
+
+          if(Model%do_mynnedmf .or.                                 & 
+                           Model%imfdeepcnv == Model%imfdeepcnv_gf ) then ! MYNN PBL or GF conv
+              !-- MYNN PBL or convective GF
+              !-- use cloud fractions with SGS clouds
+              do k=1,lmk
+                do i=1,im
+                  clouds(i,k,1)  = clouds1(i,k)
+                enddo
+              enddo
+
+                ! --- use clduni as with the GFDL microphysics.
+                ! --- make sure that effr_in=.true. in the input.nml!
+                call progclduni (plyr, plvl, tlyr, tvly, ccnd, ncndl,   & !  ---  inputs
+                         Grid%xlat, Grid%xlon, Sfcprop%slmsk, dz,delp,  &
+                         IM, LMK, LMP, clouds(:,1:LMK,1),               &
+                         effrl, effri, effrr, effrs, Model%effr_in ,    &
+                         clouds, cldsa, mtopa, mbota, de_lgth)            !  ---  outputs
+
+          else
+            ! MYNN PBL or GF convective are not used
+            call progcld6 (plyr,plvl,tlyr,qlyr,qstl,rhly,tracer1,   & !  --- inputs
```

Comment:
Hi @mzhangw @climbfuji  @ChunxiZhang-NOAA 

I changed 
call progcld5 (plyr,plvl,tlyr,qlyr,qstl,rhly,tracer1,   & !  --- inputs
into
call progcld6 (plyr,plvl,tlyr,qlyr,qstl,rhly,tracer1,   & !  --- inputs
during the merging and debugging.

Is this correct?

Thanks!

Bin

Reply:
I believe so, yes. progcld6 in the HAFS branch should be the same as progcld5 in develop, the new progcld5 in the HAFS branch is only for FA.